### PR TITLE
PD-972: Dast automation scenario UI and Integration

### PR DESCRIPTION
**Pull request by Yibaebi**

### CODE REVIEW

File: app/components/project-settings/view-scenario/parameter-item/index.hbs [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    @showHeader={{true}}
+    @blurOverlay={{true}}
+    @noGutter={{true}}
+  >
+    <:default>
+      <AkStack
+        @direction='column'
+        @spacing='2'
+        class='p-3 pr-6'
+        {{style minWidth='423px'}}
```
Comment (future-pirate-king) [inline app/components/project-settings/view-scenario/parameter-item/index.hbs:131]: can we round this to 420px

File: app/styles/_color.scss [hunk 2]
```diff
@@ -95,10 +95,21 @@ body {
...
 
+  // for ak-autocomplete component
+  --zIndex-clear-btn: 1501;
+
   // special case variable for home-page/organization-dashboard/side-nav
   --side-menu-active-background-dark: #303339;
   --side-menu-active-background-light: rgba(254, 77, 63, 0.1);
 
+  // special case variable for project-settings
+  --project-settings-light-background-color: #fbfbfb;
```
Comment (future-pirate-king) [inline app/styles/_color.scss:106]: use `--background-light`

File: app/styles/_color.scss [hunk 3]
```diff
@@ -95,10 +95,21 @@ body {
   --zIndex-snackbar: 1400;
   --zIndex-tooltip: 1500;
 
+  // for ak-autocomplete component
+  --zIndex-clear-btn: 1501;
```
Comment (future-pirate-king) [inline app/styles/_color.scss:99]: remove this

File: app/styles/_color.scss [hunk 4]
```diff
@@ -95,10 +95,21 @@ body {
...
+
   // special case variable for home-page/organization-dashboard/side-nav
   --side-menu-active-background-dark: #303339;
   --side-menu-active-background-light: rgba(254, 77, 63, 0.1);
 
+  // special case variable for project-settings
+  --project-settings-light-background-color: #fbfbfb;
+
+  // special case variable for dast scenario parameter item
+  --dast-scenario-parameter-box-shadow: -1px -1px 0px 0px #f8f8f8 inset,
```
Comment (future-pirate-king) [inline app/styles/_color.scss:109]: replace this with border using standard colors

File: mirage/factories/organization.js
Comment (future-pirate-king) [inline mirage/factories/organization.js:1]: discard changes

File: app/components/project-settings/view-scenario/index.hbs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            data-test-projectSettings-viewScenario-paramListEmptyDescText
+          >
+            {{t 'dastAutomation.noParamaterAvailable'}}
+          </AkTypography>
+        </AkStack>
+      </AkStack>
+
+    {{else}}
+      <AkStack @direction='column' @width='full'>
+        {{#each this.parameterList key='id' as |pm|}}
```
Comment (future-pirate-king) [inline app/components/project-settings/view-scenario/index.hbs:57]: remove this `key`

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

File: app/components/project-settings/general-settings/dynamicscan-automation-settings/scenario/index.hbs [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+      >
+        <AkStack @width='full' @spacing='4' @alignItems='center'>
+          <AkTypography
+            @fontWeight='regular'
+            @color='neutral'
+            {{style minWidth='75px'}}
+            data-test-genSettings-dynScanAutoSettings-scenarioAddModal-scenarioNameTextfieldTitle
+          >
+            {{t 'scenario'}}
+          </AkTypography>
```
[the remark this answers is not in the document]
Comment (Yibaebi) [reply app/components/project-settings/general-settings/dynamicscan-automation-settings/scenario/index.hbs:77]: There's no label argument in `AkTexrField`. Do you mean to make the tag a `label`?

File: app/components/project-settings/general-settings/dynamicscan-automation-settings/scenario/index.hbs [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+  @reloadScenarioList={{this.reloadProjectScenarios}}
+/>
+
+{{#if this.showAddScenarioModal}}
+  <AkModal
+    @headerTitle='{{t "dastAutomation.addScenario"}}'
+    @onClose={{this.closeAddScenarioModal}}
+    @onClickOverlay={{this.closeAddScenarioModal}}
+    @showHeader={{true}}
+    @blurOverlay={{true}}
```
[the remark this answers is not in the document]
Comment (Yibaebi) [reply app/components/project-settings/general-settings/dynamicscan-automation-settings/scenario/index.hbs:58]: Nope. Just a personal preference. Is there a problem with this?

File: app/components/project-settings/view-scenario/add-parameter-form/index.hbs [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+          @arrow={{true}}
+          @disabled={{not this.changeset.parameterType}}
+          data-test-projectSettings-viewScenario-inputTypeTextField-errorTooltip
+        >
+          <:tooltipContent>
+            <div local-class='tooltip-content'>
+              <AkTypography
+                @variant='body2'
+                @color='inherit'
+                {{style whiteSpace='wrap' minWidth='220px'}}
```
[the remark this answers is not in the document]
Comment (Yibaebi) [reply app/components/project-settings/view-scenario/add-parameter-form/index.hbs:28]: It's actually not.

File: mirage/scenarios/default.js
Comment (future-pirate-king) [inline mirage/scenarios/default.js:1]: discard these changes

File: mirage/config.js
Comment (future-pirate-king) [inline mirage/config.js:1]: discard these changes

**Review by future-pirate-king — CHANGES_REQUESTED** [review]

**Review by future-pirate-king — APPROVED** [review]

**Pull request merged**